### PR TITLE
Replace `--disallow-any` flags with separate boolean flags.

### LINKED ISSUE #4089: Replace `--disallow-any` flags with separate boolean flags.
The `--disallow-any` flags interact in unexpected ways with other flags and they make configuration more complicated. We should replace them with a set of simple boolean flags. See discussion in #4076.

### PR DESCRIPTION
**Pull request by ilinum**
This PR also gets rid of `--disallow-any=unannotated` because it
is the same thing as `--disallow-untyped-defs`.

Fix #4089

### CODE REVIEW

File: mypy/main.py [hunk 1]
```diff
@@ -272,10 +254,18 @@ def add_invertible_flag(flag: str,
...
     parser.add_argument('--follow-imports', choices=['normal', 'silent', 'skip', 'error'],
                         default='normal', help="how to treat imports (default normal)")
-    parser.add_argument('--disallow-any', type=disallow_any_argument_type, default=[],
-                        metavar='{{{}}}'.format(', '.join(disallow_any_options)),
-                        help="disallow various types of Any in a module. Takes a comma-separated "
-                             "list of options (defaults to all options disabled)")
+    parser.add_argument('--disallow-any-unimported', default=False, action='store_true',
+                        help="disallow usage of types that come from unfollowed imports")
+    parser.add_argument('--disallow-any-expr', default=False, action='store_true',
+                        help='disallow all expressions in the module that have type Any')
```
Comment (gvanrossum) [inline mypy/main.py]: I'd drop "in the module" since that's implicit (and you don't say it for the other flags).

File: mypy/main.py [hunk 2]
```diff
@@ -272,10 +254,18 @@ def add_invertible_flag(flag: str,
                         help="silently ignore imports of missing modules")
     parser.add_argument('--follow-imports', choices=['normal', 'silent', 'skip', 'error'],
                         default='normal', help="how to treat imports (default normal)")
-    parser.add_argument('--disallow-any', type=disallow_any_argument_type, default=[],
-                        metavar='{{{}}}'.format(', '.join(disallow_any_options)),
-                        help="disallow various types of Any in a module. Takes a comma-separated "
-                             "list of options (defaults to all options disabled)")
+    parser.add_argument('--disallow-any-unimported', default=False, action='store_true',
+                        help="disallow usage of types that come from unfollowed imports")
```
Comment (gvanrossum) [inline mypy/main.py]: Maybe "disallow Any types resulting from unfollowed imports" ?

Comment (ilinum) [reply mypy/main.py]: Yep, this seems reasonable! Done!

File: mypy/main.py (hunk 1 above)
Comment (ilinum) [reply mypy/main.py]: Yep!

**Comment by gvanrossum** [comment]
Thanks! Now I have a whole bunch of internal mypy.ini files to clean up...

**Pull request merged**